clamd/clamd.c

```diff
@@ -568,6 +568,12 @@ int main(int argc, char **argv)
                 ret = 1;
                 break;
             }
+
+            if (stat(CL_ENGINE_TMPDIR, &sb) != 0 && !S_ISDIR(sb.st_mode)) {
```

Comment:
In addition to swithcing to `opt->strarg`, we should probably use `CLAMSTAT` instead of `stat`. 
```suggestion
            if (CLAMSTAT(opt->strarg, &sb) != 0 && !S_ISDIR(sb.st_mode)) {
```

Reply:
You will need to also declare the `sb` variable at the top of this block, like:
```
        STATBUF sb;
```